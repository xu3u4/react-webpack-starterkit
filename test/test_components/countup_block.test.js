import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react'

import CountupBlock from 'components/countup_block';

describe('<CountupBlock>', () => {
  afterEach(cleanup);

  it ('click to change number', () => {
    const { container, getByText, getByTestId } = render(<CountupBlock />)
    expect(getByTestId('number_block').textContent).toEqual("0");
    fireEvent.click(getByText('Click'))
    expect(getByTestId('number_block').textContent).toEqual("1");
  });
});