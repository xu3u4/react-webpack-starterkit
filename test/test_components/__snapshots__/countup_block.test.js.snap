// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CountupBlock> render components 1`] = `
<div
  className="countup"
>
  <div
    className="number_block plain"
  >
    3
  </div>
  <button
    className="simple_btn"
    onClick={[Function]}
    type="button"
  >
    Click
  </button>
</div>
`;
